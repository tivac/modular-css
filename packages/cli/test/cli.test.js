"use strict";

const tester = require("cli-tester");

const read = require("@modular-css/test-utils/read.js")(__dirname);

const cli = require.resolve("../cli.js");

describe("/cli.js", () => {
    afterAll(() => require("shelljs").rm("-rf", "./packages/cli/test/output"));

    it("should show help with no args", async () => {
        const out = await tester(
            cli
        );

        expect(out.code).toBe(2);
        expect(out.stdout).toMatchSnapshot();
    });

    it("should default to outputting to stdout", async () => {
        const out = await tester(
            cli,
            "./packages/cli/test/specimens/simple.css"
        );

        expect(out.code).toBe(0);
        expect(out.stdout).toMatchSnapshot();
    });

    it("should support outputting to a file (--out)", async () => {
        const out = await tester(
            cli,
            "--out=./packages/cli/test/output/simple.css",
            "./packages/cli/test/specimens/simple.css"
        );

        expect(out.code).toBe(0);
        expect(read("simple.css")).toMatchSnapshot();
    });

    it("should support outputting compositions to a file (--json)", async () => {
        const out = await tester(
            cli,
            "--json=./packages/cli/test/output/classes.json",
            "./packages/cli/test/specimens/simple.css"
        );

        console.log(out.stdout);
        console.log(out.stderr);

        expect(out.code).toBe(0);
        expect(read("classes.json")).toMatchSnapshot();
    });

    it("should return the correct error code on invalid CSS", async () => {
        const out = await tester(
            cli,
            "./packages/cli/test/specimens/invalid.css"
        );

        expect(out.code).toBe(1);
        expect(out.stderr).toMatch("Invalid composes reference");
    });

    it("should support disabling url() rewriting (--no-rewrite)", async () => {
        const out = await tester(
            cli,
            "--no-rewrite",
            "--out=./packages/cli/test/output/no-rewrite.css",
            "./packages/cli/test/specimens/no-rewrite.css"
        );

        expect(out.code).toBe(0);
        expect(read("no-rewrite.css")).toMatchSnapshot();
    });
});
