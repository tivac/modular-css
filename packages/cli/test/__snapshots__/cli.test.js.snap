// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/cli.js should default to outputting to stdout 1`] = `
"/* packages/cli/test/specimens/simple.css */
.mc73666a66_wooga { color: red; }"
`;

exports[`/cli.js should show help with no args 1`] = `
"CLI for modular-css

  Usage
  $ modular-css [options] <glob>...

  Options
  --dir,  -d <dir>    Directory to search from [process cwd]
  --out,  -o <file>   File to write output CSS to [stdout]
  --json, -j <file>   File to write output compositions JSON to
  --map,  -m          Include inline source map in output
  --help              Show this help"
`;

exports[`/cli.js should support outputting compositions to a file 1`] = `
"{
    \\"packages/cli/test/specimens/simple.css\\": {
        \\"wooga\\": \\"mc73666a66_wooga\\"
    }
}"
`;

exports[`/cli.js should support outputting to a file 1`] = `
"/* packages/cli/test/specimens/simple.css */
.mc73666a66_wooga { color: red; }
"
`;
